### PR TITLE
Add animation callbacks

### PR DESCRIPTION
**Pull request by mjurfest**
Add callbacks for the animation
- onEnter
- onEntering
- onEntered
- onExit
- onExiting
- onExited

### CODE REVIEW

File: src/components/Animator/README.md [hunk 1]
```diff
@@ -11,6 +11,12 @@
 | className | string | false | - | The class will be added as a **direct parent class** for all the children it will have|
 | debug | string | false | enter, entering, exit, exiting | Emulates the real animation classes and styles without the need to animate. Just exit your element visible and add to debug different stages. enter: 'enter', entering: 'enter enter-active', exit: 'exit', exiting: 'exit exit-active' |
 | dataHook | string | - | - | will convert to data-hook |
+| onEnter | function | - | - | Callback onEnter  |
+| onEntering | function | - | - | Callback onEntering  |
+| onEntered | function | - | - | Callback onEntered  |
+| onExit | function | - | - | Callback onExit  |
+| onExiting | function | - | - | Callback onExiting  |
+| onExitedAnimation | function | - | - | Callback onExitedAnimation  |
```
Comment (egidijusja) [inline src/components/Animator/README.md:19]: I think it is enough to have `onExited`, we don't need Animation postfix, to keep it consistent with other props (onEnter, onEntered, etc..)

File: src/components/Animator/components/CSSTransitionWrapper.js [hunk 2]
```diff
@@ -57,12 +61,18 @@ class CSSTransitionWrapper extends React.Component {
       case 'entering':
         update = {enter: true, entering: true};
         break;
+      case 'entered':
+        update = {enter: true, entering: true, entered: true};
+        break;
```
Comment (egidijusja) [inline src/components/Animator/components/CSSTransitionWrapper.js:66]: I looked through the code and I cannot find any usage of entered and exited states. I think we don't need `case 'entered'` and `case 'exited'`

**Review by egidijusja — CHANGES_REQUESTED** [review]

File: src/components/Animator/README.md (hunk 1 above)
Comment (mjurfest) [reply src/components/Animator/README.md]: I have changed all names

**Review by egidijusja — APPROVED** [review]
Looks good!

**Review by mykas — APPROVED** [review]

**Pull request merged**